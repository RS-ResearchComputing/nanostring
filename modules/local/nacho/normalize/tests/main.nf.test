nextflow_process {

    name "Test Process NACHO_NORMALIZE"
    script "../main.nf"
    process "NACHO_NORMALIZE"
    tag "modules"

    test("Should run without failures") {

        when {
            process {
                """
                // RCC Files: Collect from sample sheet
                input[0] = Channel.fromPath(params.pipelines_testdata_base_path + 'nanostring/samplesheets/samplesheet_test.csv', checkIfExists: true)
                    .splitCsv(header: true)
                    .map { row -> row.RCC_FILE } // Select first column: path to file
                    .collect()

                // Sample sheet
                input[1] = file(params.pipelines_testdata_base_path + 'nanostring/samplesheets/samplesheet_test.csv', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
