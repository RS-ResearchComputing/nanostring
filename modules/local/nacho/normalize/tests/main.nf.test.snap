{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "normalized_counts.tsv:md5,a0124c7a24bd04296f441d9ade82a05f"
                ],
                "1": [
                    "normalized_counts_wo_HKnorm.tsv:md5,5a2ce112c24e1b0d0f4cf3392111ef9e"
                ],
                "2": [
                    "versions.yml:md5,0b8556f7dd763f21894470ecc737ca3b"
                ],
                "normalized_counts": [
                    "normalized_counts.tsv:md5,a0124c7a24bd04296f441d9ade82a05f"
                ],
                "normalized_counts_wo_HK": [
                    "normalized_counts_wo_HKnorm.tsv:md5,5a2ce112c24e1b0d0f4cf3392111ef9e"
                ],
                "versions": [
                    "versions.yml:md5,0b8556f7dd763f21894470ecc737ca3b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-19T19:00:09.938938106"
    }
}